src/coreclr/gc/gc.cpp

```diff
@@ -52052,6 +52052,34 @@ void GCHeap::DiagWalkHeap (walk_fn fn, void* context, int gen_number, bool walk_
     gc_heap::walk_heap (fn, context, gen_number, walk_large_object_heap_p);
 }
 
+void GCHeap::DiagWalkHeapStandalone (walk_fn fn, void* context, int gen_number, bool walk_large_object_heap_p)
```

Comment:
as I mentioned when we chatted, I was hoping we'd come up with a name better than Standalone, especially now since we don't do suspend/restart EE anymore in this, maybe just call it something like DiagWalkHeapWithACHandling. feel free to suggest something better.

also I would add a comment that says the caller is responsible for suspending the EE before callling this.

Reply:
I thought about it but I didn't come up with any name suggestion that felt better. One convention used elsewhere in profiler code is that we call methods "Async" when they can be called at arbitrary times vs. "Synchronous" when they can only be called from within profiler callbacks. Its overloaded terminology but applies here if you wanted to use it.

I'm just fine calling this DiagWalkHeapWithACHandling or any other name folks would like.

Reply:
What are the chances that you will want to add another flag to this method? Would it make sense for this method to have more general name and take flags to control the specific behavior?

```
typedef enum
{
    walk_heap_large_object_heap = 1,
    walk_heap_update_allocation_contexts = 2,
} walk_heap_type;
```

```
void GCHeap::DiagWalkHeap2 (walk_fn fn, void* context, int gen_number, walk_heap_type type)
```

Reply:
> What are the chances that you will want to add another flag to this method?

I'm thinking that it is low if there aren't any plans to add more behaviors for how the gc heap is walked
The way I see it, this method spans a larger scope of the 5 steps needed to walk the gc heap compared to `GCHeap::DiagWalkHeap` which is solely called within a GC and assumes the other steps have/will happen.
```
Suspending EE
    Fix all heap allocation contexts            | (This method's scope)
        Walking GC Heap (DiagWalkHeap's scope)  | 
    Repair all heap allocation contexts         |
Resuming EE
```
There was the thought of including the EE suspend/resume in this method's scope, but initially I felt that leaving the responsibility of EE suspend/resume to the ICorProfiler API was simpler because:
1. all other EE suspensions in gc.cpp are [GCToEEInterface::SuspendEE](https://github.com/dotnet/runtime/blob/c241cc99ac0cb3862bdee5e6f931f0ba60537631/src/coreclr/vm/gcenv.ee.cpp#L38-L52), whose assertions aren't appropriate since this method doesn't actually perform a GC
2. Other ICorProfiler API's already [suspend EE](https://github.com/dotnet/runtime/blob/c241cc99ac0cb3862bdee5e6f931f0ba60537631/src/coreclr/vm/proftoeeinterfaceimpl.cpp#L6837-L6868)/[resume EE ](https://github.com/dotnet/runtime/blob/c241cc99ac0cb3862bdee5e6f931f0ba60537631/src/coreclr/vm/proftoeeinterfaceimpl.cpp#L6870-L6900), so there's precedence + logic accounting for an already suspended EE makes sense.

Re: Naming, I also didn't come up with any better names (`DiagWalkHeapOutsideOfGC` didn't feel better than `DiagWalkHeapStandalone`), but `DiagWalkHeapWithACHandling` seems more appropriate especially considering the scoping above.

I'll add comments that EE Suspension/Resumption responsibility falls to callers of `GCHeap::DiagWalkHeapWithACHandling`.